dev/tools/svg2dart/lib/svg2dart.dart

```diff
@@ -0,0 +1,547 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+import 'dart:math';
+
+import 'package:collection/collection.dart';
+import 'package:meta/meta.dart';
+import 'package:vector_math/vector_math_64.dart';
+import 'package:xml/xml.dart' as xml show parse;
+import 'package:xml/xml.dart' hide parse;
+
+// String to use for a single indentation.
+const String kIndent = '  ';
+
+/// Represents an entire animation.
+class Animation {
+  const Animation(this.size, this.paths);
+
+  factory Animation.fromFrameData(List<FrameData> frames) {
+    _validateFramesData(frames);
+    final Point<double> size = frames[0].size;
+    final List<PathAnimation> paths = <PathAnimation>[];
+    for (int i = 0; i < frames[0].paths.length; i += 1) {
+      paths.add(new PathAnimation.fromFrameData(frames, i));
+    }
+    return new Animation(size, paths);
+  }
+
+  /// The size of the animation (width, height) in pixels.
+  final Point<double> size;
+
+  /// List of paths in the animation.
+  final List<PathAnimation> paths;
+
+  static void _validateFramesData(List<FrameData> frames) {
+    final Point<double> size = frames[0].size;
+    final int numPaths = frames[0].paths.length;
+    for (int i = 0; i < frames.length; i += 1) {
+      final FrameData frame = frames[i];
+      if (size != frame.size)
+        throw new Exception(
+            'All animation frames must have the same size, '
+                'first frame size was: (${size.x}, ${size.y}) '
+                'frame $i size was: (${frame.size.x}, ${frame.size.y}) '
+        );
+      if (numPaths != frame.paths.length)
+        throw new Exception(
+            'All animation frames must have the same number of paths, '
+                'first frame has $numPaths paths'
```

Comment:
bonus points for removing the `s` if `numPaths` is 1 :-)
(similar on next line)

Reply:
IMO in this case code simplicity is more important than not having this exception say '1 paths'